src/main/java/org/junitpioneer/jupiter/TemporaryDirectory.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2016-2021 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junitpioneer.jupiter;
+
+import java.io.IOException;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+
+public final class TemporaryDirectory implements ResourceFactory<Path> {
+
+	private final String prefix;
+
+	public TemporaryDirectory() {
+		this.prefix = "";
+	}
+
+	public TemporaryDirectory(String prefix) {
+		this.prefix = prefix;
+	}
+
+	@Override
+	public Resource<Path> create() throws Exception {
+		return new Resource<Path>() {
+
+			private Path tempDir;
+
+			@Override
+			public Path get() throws Exception {
+				return (tempDir = Files.createTempDirectory(prefix));
```

Comment:
Doesn't this reassign a new `tempDir` every time `get()` is called? Generally speaking, I'd prefer to have a final field, whose value gets assigned on creation.

Reply:
It does. I'm assumed that `Resource::get` will be called at most once by the extension and will never be called by user code.

I drove this code in a "minimalist" TDD fashion, so thanks for pointing out the flaw! It's now in the TODO list.